src/tools/optimization-options.h

```diff
@@ -42,27 +42,34 @@ struct OptimizationOptions : public Options {
                   passOptions.optimizeLevel = 0;
                   passOptions.shrinkLevel = 0;
                 })
-           .add("", "-O1", "execute -O1 optimization passes",
+           .add("", "-O1", "execute -O1 optimization passes (quick&useful opts, useful for iteration builds)",
                 Options::Arguments::Zero,
                 [this](Options*, const std::string&) {
                   passOptions.optimizeLevel = 1;
                   passOptions.shrinkLevel = 0;
                   passes.push_back(DEFAULT_OPT_PASSES);
                 })
-           .add("", "-O2", "execute -O2 optimization passes",
+           .add("", "-O2", "execute -O2 optimization passes (most opts, generally gets most perf)",
                 Options::Arguments::Zero,
                 [this](Options*, const std::string&) {
                   passOptions.optimizeLevel = 2;
                   passOptions.shrinkLevel = 0;
                   passes.push_back(DEFAULT_OPT_PASSES);
                 })
-           .add("", "-O3", "execute -O3 optimization passes",
+           .add("", "-O3", "execute -O3 optimization passes (spends potentially a lot of time optimizing)",
                 Options::Arguments::Zero,
                 [this](Options*, const std::string&) {
                   passOptions.optimizeLevel = 3;
                   passOptions.shrinkLevel = 0;
                   passes.push_back(DEFAULT_OPT_PASSES);
                 })
+           .add("", "-O4", "execute -O4 optimization passes (also flatten the IR and do flat-dependent opts)",
```

Comment:
The other comments give a good heuristic to the user, whereas this one is more descriptive of what it makes Binaryen do. I'd word this more like "optimizations that usually aren't worth it, can be many times slower than -O3" to give the user a better sense of what the tradeoff is here (optimization at all costs)

Reply:
Good point, I'll improve this.